jdaviz/configs/default/plugins/gaussian_smooth/tests/test_gaussian_smooth.py

```diff
@@ -68,21 +65,109 @@ def test_linking_after_spectral_smooth(spectrum1d_cube):
     assert dc.external_links[2].cids1[0] == dc[0].pixel_component_ids[2]
     assert dc.external_links[2].cids2[0] == dc[-1].pixel_component_ids[2]
 
+    # Mouseover should automatically jump from one spectrum
+    # to another, depending on which one is closer.
+
+    spec_viewer.on_mouse_or_key_event({'event': 'mousemove', 'domain': {'x': 4.633e-7, 'y': 60}})
+    assert spec_viewer.label_mouseover.pixel == 'x=01.0'
+    assert spec_viewer.label_mouseover.world_label_prefix == 'Wave'
+    assert spec_viewer.label_mouseover.world_ra == '4.62360e-07'
+    assert spec_viewer.label_mouseover.world_dec == 'm'
+    assert spec_viewer.label_mouseover.world_label_prefix_2 == 'Flux'
+    assert spec_viewer.label_mouseover.world_ra_deg == '9.20000e+01'
+    assert spec_viewer.label_mouseover.world_dec_deg == 'Jy'
+    assert spec_viewer.label_mouseover.icon == 'a'
+
+    spec_viewer.on_mouse_or_key_event({'event': 'mousemove', 'domain': {'x': 4.633e-7, 'y': 20}})
+    assert spec_viewer.label_mouseover.pixel == 'x=01.0'
+    assert spec_viewer.label_mouseover.world_label_prefix == 'Wave'
+    assert spec_viewer.label_mouseover.world_ra == '4.62360e-07'
+    assert spec_viewer.label_mouseover.world_dec == 'm'
+    assert spec_viewer.label_mouseover.world_label_prefix_2 == 'Flux'
+    assert spec_viewer.label_mouseover.world_ra_deg == '1.47943e+01'
+    assert spec_viewer.label_mouseover.world_dec_deg == 'Jy'
+    assert spec_viewer.label_mouseover.icon == 'b'
+
+    # Check mouseover behavior when we hide everything.
+    for lyr in spec_viewer.layers:
+        lyr.visible = False
+
+    spec_viewer.on_mouse_or_key_event({'event': 'mousemove', 'domain': {'x': 4.633e-7, 'y': 60}})
+    assert spec_viewer.label_mouseover.pixel == ''
+    assert spec_viewer.label_mouseover.world_label_prefix == '\xa0'
+    assert spec_viewer.label_mouseover.world_ra == ''
+    assert spec_viewer.label_mouseover.world_dec == ''
+    assert spec_viewer.label_mouseover.world_label_prefix_2 == '\xa0'
+    assert spec_viewer.label_mouseover.world_ra_deg == ''
+    assert spec_viewer.label_mouseover.world_dec_deg == ''
+    assert spec_viewer.label_mouseover.icon == ''
+
 
-@pytest.mark.filterwarnings("ignore::UserWarning")
```

Comment:
I moved the warning check to the affected line. This way, we won't accidentally ignore warning that we're not supposed to ignore.